ksql-engine/src/main/java/io/confluent/ksql/analyzer/Analyzer.java

```diff
@@ -199,8 +199,7 @@ private void analyzeNonStdOutSink(final Sink sink) {
     }
 
     private KeyFormat buildKeyFormat() {
-      final Optional<KsqlWindowExpression> ksqlWindow = Optional
-          .ofNullable(analysis.getWindowExpression())
+      final Optional<KsqlWindowExpression> ksqlWindow = analysis.getWindowExpression()
           .map(WindowExpression::getKsqlWindowExpression);
 
       return ksqlWindow
```

Comment:
Shouldn't there be an ifPresent here since the WindowExpression is Optional?

Reply:
`Optional.map` deals with that internally - if analysis returns an empty `Optional`, then `ksqlWindow` will also be empty.